### PR TITLE
[fmt] Update to 9.0.0

### LINKED ISSUE #25642: [fmt] update to 9.0.0
Library name: fmt

New version number: 9.0.0

Other information that may be useful (release notes, etc...)

https://github.com/fmtlib/fmt/releases/tag/9.0.0

### PR DESCRIPTION
**Pull request by spnda**
**Describe the pull request**
Updates the fmtlib port to 9.0.0.

- #### What does your PR fix?
  Fixes #25642.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All, No, not required

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by spnda** [comment]
It seems like the `openimageio` port depends on `fmt` and fails to build with fmt 9.0.0. Is there perhaps some functionality to restrict building openimageio with fmt 9.0.0? I personally can't test out builds or see where and why it fails because the bundled meson doesn't seem to like macOS 13 and Xcode 14.

**Comment by dg0yt** [comment]
In https://github.com/microsoft/vcpkg/pull/23918, I arrived at updating openimageio. But the PR isn't ready yet.

**Comment by FrankXie05** [comment]
Log from x86-windows
```
D:\buildtrees\openimageio\x86-windows-dbg\include\OpenImageIO\detail\fmt\core.h(1733): error C2338: static_assert failed: 'Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt'
D:\buildtrees\openimageio\x86-windows-dbg\include\OpenImageIO\detail\fmt\core.h(1753): note: see reference to function template instantiation 'fmt::v9::detail::value<Context> fmt::v9::detail::make_value<Context,const T&>(const OpenImageIO_v2_3::string_view&)' being compiled
        with
        [
            Context=context,
            T=OpenImageIO_v2_3::string_view
        ]
````
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/9080644/install-x86-windows-dbg-out.2.log)

**Comment by BullyWiiPlaza** [comment]
I'm hoping for a merge soon since I was also getting these annoying warnings lately in Visual Studio 2022:
`C:\vcpkg\installed\x64-windows-static\include\fmt\format.h(1812,18): warning C4189: 'zero': local variable is initialized but not referenced`
It seems like fmt version 9.0.0 fixes it.

**Comment by spnda** [comment]
@BullyWiiPlaza I'm also eager to get this merged.

The CI failures currently only come from the openimageio port being severely outdated and it currently will not compile with fmt v9. Though, as I'm not too experienced with how vcpkg handles port dependencies, I would like to ask the vcpkg team if there was any way in telling the CI that that port simply won't compile with fmt 9, and have the compile failures be expected? That would get this PR merged, while having openimageio users still be able to use it with the previous version of the fmt port.

**Comment by FrankXie05** [comment]
@spnda @BullyWiiPlaza Sorry, we have to wait for `Openimageio` to update before we can update `fmt` to 9.0.0. However, you can temporarily clone the PR #23918 changes to self-update `Openimageiolo` locally to support `fmt` 9.0.0. :)

**Comment by walge-laine** [comment]
Now that #23918 has been merged, is this PR being worked on?

**Comment by spnda** [comment]
@walge-laine There's nothing for this PR to work on anymore. I'll just trigger the CI to re-run and all of that should succeed. Thanks for reminding me though, I was not subscribed to the other PR.

**Review by FrankXie05 — DISMISSED** [review]

**Comment by FrankXie05** [comment]
@spnda @walge-laine  I merged the branch of this pr into the master. Now this PR is working fine, I think we can merge it now. :)

File: ports/fmt/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fmt",
-  "version": "8.1.1",
+  "version": "9.0.0",
   "port-version": 2,
```
Comment (Thomas1664) [inline ports/fmt/vcpkg.json:4]: Please remove `port-version`.

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**